compiler/rustc_infer/src/infer/outlives/verify.rs

```diff
@@ -245,10 +245,19 @@ impl<'cx, 'tcx> VerifyBoundCx<'cx, 'tcx> {
                     "declared_generic_bounds_from_env_for_erased_ty: region_bound_pair = {:?}",
                     (r, p)
                 );
+                // Fast path for the common case.
+                match (erased_ty.kind(), &p) {
+                    (ty::Alias(_, a1), GenericKind::Alias(a2)) if a1.def_id != a2.def_id => {
+                        return None;
+                    }
+                    (ty::Param(p1), GenericKind::Param(p2)) if p1 != p2 => return None,
+                    _ => {}
```

Comment:
```suggestion
                    (ty::Alias(_, a1), GenericKind::Alias(a2)) if a1.def_id == a2.def_id =>{}
                    (ty::Param(p1), GenericKind::Param(p2)) if p1 == p2 => {}
                    _ => return None,
```
This should cover more cases.

Reply:
This is missing the case of `GenericKind::Placeholder`.